### PR TITLE
Panzer: implemet a method to add IOSS information records to the output

### LINKED ISSUE #9436: Panzer: Add IOSS information records to the output
## Enhancement
@trilinos/panzer 

Develop a way for Panzer clients to add Information Records to their IOSS output.

### PR DESCRIPTION
**Pull request by tkordenbrock**
Implement an addInformationRecords() method that adds a vector of strings
to the Panzer STK Interface.  The strings are depuped as they are added
and then added to the STK mesh during setupExodusFile().

@trilinos/panzer 

* Closes #9436 

## Testing

addInformationRecords() is tested by PanzerAdaptersSTK_tSTK_IO.exe.

### CODE REVIEW

File: packages/panzer/adapters-stk/test/stk_interface_test/tSTK_IO.cpp [hunk 1]
```diff
@@ -237,6 +238,59 @@ TEUCHOS_UNIT_TEST(tSTK_IO, transient_fields)
...
+   mesh->addInformationRecords(info_records_2);
+   mesh->addInformationRecords(info_records_3);
+   mesh->addInformationRecords(info_records_4);
+   mesh->addInformationRecords(info_records_5);
+   mesh->addInformationRecords(info_records_6);
+   factory.completeMeshConstruction(*mesh,MPI_COMM_WORLD);
+
+   TEST_ASSERT(mesh!=Teuchos::null);
+
+   mesh->writeToExodus("info_records.exo");
```
Comment (rppawlo) [inline packages/panzer/adapters-stk/test/stk_interface_test/tSTK_IO.cpp:291]: This test just writes the file, but it doesn't check that the info was actually written. Should it open the exodus file, read out the strings and check that the info is correct?

**Review by rppawlo — APPROVED** [review]
Looks good to me. Just had one question about the unit test.

File: packages/panzer/adapters-stk/test/stk_interface_test/tSTK_IO.cpp (hunk 1 above)
Comment (tkordenbrock) [reply packages/panzer/adapters-stk/test/stk_interface_test/tSTK_IO.cpp:296]: Good point.  I pushed an update that checks that the info records were written to the Exodus file.

**Review by rppawlo — APPROVED** [review]
Thanks @tkordenbrock !

**Pull request merged**